packages/terra-spacer/src/Spacer.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { withBps } from 'react-bps';
+import 'terra-base/lib/baseStyles';
+import styles from './Spacer.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Child Nodes.
+   */
+  children: PropTypes.node,
+  /**
+   * Sets top margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginTop: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets bottom margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginBottom: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets left margin One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginLeft: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets right margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  marginRight: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets top padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingTop: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets bottom padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingBottom: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets left padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingLeft: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets right padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  paddingRight: PropTypes.oneOf(['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4']),
+  /**
+   * Sets the display to be inline-block.
+   */
+  isInlineBlock: PropTypes.bool,
+  /**
+   * Object used to determine at what window size props should be active. e.g { 768: { marginBottom: 'small-2' }}
+   */
+  bps: PropTypes.object,
```

Comment:
You can define this like:
```
const basePropTypes = {
  // what you have, minus bps
}

const bpsPropType = PropTypes.objectOf(PropTypes.shape(basePropTypes);

const propTypes = Object.assign({ bps: bpsPropType }, basePropTypes);
```

Reply:
@tbiethman So bps here is a predefined object of react-bps package. (find more info here https://www.npmjs.com/package/react-bps). react-bps already has bps PropType validation built in which will not allow consumers to use strings or any other data types instead of numbers for specifying the breakpoints of view port sizes.